src/test/java/org/apache/commons/lang3/EnumUtilsTest.java

```diff
@@ -163,6 +165,26 @@ public void test_getEnumIgnoreCase_defaultEnum() {
         assertNull(EnumUtils.getEnumIgnoreCase(Traffic.class, "PURPLE", null));
     }
 
+    @Test
+    public void test_getEnumByPredicate() {
+        assertEquals(WithExtraData.A, EnumUtils.getEnumByPredicate(WithExtraData.class, WithExtraData::isA).get());
+        assertNotEquals(WithExtraData.A, EnumUtils.getEnumByPredicate(WithExtraData.class, WithExtraData::isNotA).get());
+        assertEquals(WithExtraData.A, EnumUtils.getEnumByPredicate(WithExtraData.class, value -> true).get());
+        assertEquals(Optional.empty(), EnumUtils.getEnumByPredicate(WithExtraData.class, value -> false));
+    }
+
+    @Test
+    public void test_getEnumByPredicate_nullInputs() {
+        assertFalse(EnumUtils.getEnumByPredicate(null, WithExtraData::isA).isPresent());
+        assertFalse(EnumUtils.getEnumByPredicate(WithExtraData.class, null).isPresent());
+    }
+
+    @Test
+    public void test_getEnumByPredicate_nonEnumClass() {
+        final Class rawType = Object.class;
+        assertFalse(EnumUtils.getEnumByPredicate(rawType, null).isPresent());
```

Comment:
This test does not cover checking the class' enum aspect because the predicate is null.

Reply:
Thanks, had it named differently before, but decided to align the naming with the other enum tests 🤦

Reply:
Ah, I misread the comment. Yes you're right, the test is faulty.